scripts/pdf_utils.py

```
@@ -32,12 +36,12 @@ def get_paddle_engine():
         with _ENGINE_LOCK:
             if _PADDLE_ENGINE is None:
-                print("[AI] Initializing PaddleOCR engine (first time only)...")
+                logger.info("Initializing PaddleOCR engine (first time only)...")
                 try:
                     # Deferred imports to prevent boot crashes on low-resource environments
                     from paddleocr import PPStructure
-                except ImportError as e:
-                    print(f"[AI] Error: PaddleOCR not installed correctly. {e}")
+                except ImportError:
+                    logger.exception("PaddleOCR not installed correctly")
                     raise ImportError("PaddleOCR engine is missing. If you are on the Free Tier, it might have failed to install due to size limits.")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)